src/DayPicker.tsx

```diff
@@ -425,42 +437,52 @@ export function DayPicker(initialProps: DayPickerProps) {
-                        <components.MonthsDropdown
-                          className={classNames[UI.MonthsDropdown]}
-                          aria-label={labelMonthDropdown()}
-                          classNames={classNames}
-                          components={components}
-                          disabled={Boolean(props.disableNavigation)}
-                          onChange={handleMonthChange(calendarMonth.date)}
-                          options={dropdownMonths}
-                          style={styles?.[UI.Dropdown]}
-                          value={dateLib.getMonth(calendarMonth.date)}
-                        />
-                      ) : (
-                        <span>
-                          {formatMonthDropdown(calendarMonth.date, dateLib)}
-                        </span>
-                      )}
-                      {captionLayout === "dropdown" ||
-                      captionLayout === "dropdown-years" ? (
-                        <components.YearsDropdown
-                          className={classNames[UI.YearsDropdown]}
-                          aria-label={labelYearDropdown(dateLib.options)}
-                          classNames={classNames}
-                          components={components}
-                          disabled={Boolean(props.disableNavigation)}
-                          onChange={handleYearChange(calendarMonth.date)}
-                          options={dropdownYears}
-                          style={styles?.[UI.Dropdown]}
-                          value={dateLib.getYear(calendarMonth.date)}
-                        />
-                      ) : (
-                        <span>
-                          {formatYearDropdown(calendarMonth.date, dateLib)}
-                        </span>
-                      )}
+                      {dropdownTypes.map((type) => {
+                        const uiKey =
+                          type === "month" ? "MonthsDropdown" : "YearsDropdown";
+                        const Dropdown = components[uiKey];
+                        const dropdownLabelGetter =
+                          type === "month"
+                            ? labelMonthDropdown
+                            : labelYearDropdown;
+                        const handleChange =
+                          type === "month"
+                            ? handleMonthChange
+                            : handleYearChange;
+                        const options =
+                          type === "month" ? dropdownMonths : dropdownYears;
+                        const valueGetter =
+                          type === "month" ? dateLib.getMonth : dateLib.getYear;
+                        const formatLabel =
+                          type === "month"
+                            ? formatMonthDropdown
+                            : formatYearDropdown;
+
+                        return (
+                          <Fragment key={type}>
```

Comment:
`Fragment` is not needed, `key` can be passed to the `Dropdown` component, and `span`.